File: AndroidApp/app/src/main/java/com/ep133/sampletool/ui/theme/EP133Components.kt
package com.ep133.sampletool.ui.theme

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.ep133.sampletool.domain.model.PadChannel
import com.ep133.sampletool.ui.TestTags

/**
 * The EP-133 redesign component kit — the Compose mirror of the Claude Design system's component
 * sheet (pads, chips, group selector, status dots, buttons, section labels, app chrome). Every
 * redesigned screen composes from these so the look stays consistent. All colors come from
 * [LocalEP133Tokens]; the hardware aesthetic (flat faceplate, hard ~2dp corners, mono labels) is
 * intentionally NOT Material — Material 3 components are used only where they fit (forms, sliders).
 */

/** Small hard-corner radius used across the faceplate UI. */
private val Radius = RoundedCornerShape(3.dp)

/** Mono labels/codes (the design uses JetBrains Mono; Monospace is the safe on-device fallback). */

// ── Section label — uppercase mono eyebrow over a block ───────────────────────
@Composable
fun Ep133SectionLabel(text: String, modifier: Modifier = Modifier) {
    val t = LocalEP133Tokens.current
    Text(
        text = text.uppercase(),
        modifier = modifier,
        color = t.text3,
        fontFamily = Mono,
        fontSize = 9.sp,
        fontWeight = FontWeight.Medium,
        letterSpacing = 1.6.sp,
    )
}

// ── Status dot (teal=ok, accent=active, text3=idle) ───────────────────────────
@Composable
fun Ep133StatusDot(color: Color, size: Int = 8, modifier: Modifier = Modifier) {
    Box(modifier.size(size.dp).clip(CircleShape).background(color))
}

// ── Pad cell ──────────────────────────────────────────────────────────────────
enum class PadState { Empty, Loaded, Pressed, InScale }

/**
 * A rubber pad cell: faceplate-black face, hard corners, mono id + name. [state] drives the id
 * color and the pressed glow / in-scale teal hairline, matching the design's pad variants.
 */
@Composable
fun Ep133Pad(
    id: String,
    name: String,
    state: PadState,
    modifier: Modifier = Modifier,
    onClick: (() -> Unit)? = null,
) {
    val t = LocalEP133Tokens.current
    val idColor = when (state) {
        PadState.Empty -> Color(0xFF5D5E5F)
        PadState.Loaded -> Color(0xFFC9CACB)
        PadState.Pressed -> t.accent
        PadState.InScale -> t.live
    }
    val border = when (state) {
        PadState.InScale -> t.live
        PadState.Pressed -> t.accent
        else -> t.padEdge
    }
    val face = if (state == PadState.Pressed) t.padTop else t.padFace
    Column(
        modifier = modifier
            .clip(Radius)
            .background(face, Radius)
            .border(1.dp, border, Radius)
            .then(if (onClick != null) Modifier.clickable { onClick() } else Modifier)
            .padding(vertical = 11.dp, horizontal = 8.dp),
        verticalArrangement = Arrangement.spacedBy(3.dp),
    ) {
        Text(id, color = idColor, fontFamily = Mono, fontSize = 11.sp, fontWeight = FontWeight.Bold, letterSpacing = 0.3.sp)
        Text(name.uppercase(), color = t.padName, fontFamily = Mono, fontSize = 8.sp, letterSpacing = 0.3.sp)
    }
}

// ── Group selector chip (A / B / C / D, etc.) ─────────────────────────────────
@Composable
fun Ep133GroupChip(
    label: String,
    selected: Boolean,
    modifier: Modifier = Modifier,
    subLabel: String? = null,
    onClick: () -> Unit,
) {
    val t = LocalEP133Tokens.current
    Column(
        modifier = modifier
            .clip(Radius)
            .background(if (selected) t.accent else t.inset, Radius)
            .border(1.dp, if (selected) t.accent else t.rule, Radius)
            .clickable { onClick() }
            .padding(vertical = if (subLabel != null) 6.dp else 9.dp, horizontal = 14.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(1.dp),
    ) {
        Text(
            label,
            color = if (selected) t.onAccent else t.text2,
            fontFamily = Mono,
            fontSize = 12.sp,
            fontWeight = FontWeight.Bold,
        )
        if (subLabel != null) {
            Text(
                subLabel,
                color = if (selected) t.onAccent.copy(alpha = 0.75f) else t.text3,
                fontFamily = Mono,
                fontSize = 7.5.sp,
                letterSpacing = 0.8.sp,
                fontWeight = FontWeight.Bold,
            )
        }
    }
}

// ── Group + choke bar — the shared A/B/C/D picker + choke toggle ───────────────
/**
 * The A/B/C/D group picker plus the CHOKE GROUP toggle, as one block. Both the Loop Chopper and
 * the Kit Builder target a group and can choke it, so they share this control (identical placement
 * and styling) rather than each rolling its own. The choke toggle writes `sound.mutegroup` so pads
 * in the group cut each other off.
 */
@Composable
fun Ep133GroupChokeBar(
    group: PadChannel,
    onGroupChange: (PadChannel) -> Unit,
    chokeOn: Boolean,
    onChokeChange: (Boolean) -> Unit,
    modifier: Modifier = Modifier,
    tagFor: ((PadChannel) -> String?)? = null,
    testTagFor: ((PadChannel) -> String)? = null,
) {
    val t = LocalEP133Tokens.current
    Column(modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(7.dp)) {
        // A | B | C | D — one chip per group, equal width, each tagged with its designation.
        Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(6.dp)) {
            PadChannel.entries.forEach { ch ->
                Ep133GroupChip(
                    label = ch.name,
                    selected = group == ch,
                    modifier = Modifier.weight(1f).then(
                        testTagFor?.let { Modifier.testTag(it(ch)) } ?: Modifier,
                    ),
                    subLabel = tagFor?.invoke(ch),
                    onClick = { onGroupChange(ch) },
                )
            }
        }
        // Choke toggle — full-width row, tap anywhere to flip.
        Row(
            Modifier
                .fillMaxWidth()
                .clip(Radius)
                .background(t.inset, Radius)
                .border(1.dp, t.rule, Radius)
                .clickable { onChokeChange(!chokeOn) }
                .padding(horizontal = 12.dp, vertical = 9.dp)
                .testTag(TestTags.GROUP_CHOKE_TOGGLE),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Column(Modifier.weight(1f)) {
                Text(
                    "CHOKE GROUP",
                    fontFamily = Mono, fontSize = 10.5.sp, fontWeight = FontWeight.Bold,
                    letterSpacing = 0.6.sp, color = t.text,
                )
                Text(
                    "pads in the group cut each other off",
                    fontFamily = Mono, fontSize = 8.5.sp, color = t.text3,
                )
            }
            Box(
                Modifier
                    .clip(Radius)
                    .background(if (chokeOn) t.accent else t.chrome, Radius)
                    .padding(horizontal = 13.dp, vertical = 5.dp),
            ) {
                Text(
                    if (chokeOn) "ON" else "OFF",
                    fontFamily = Mono, fontSize = 10.sp, fontWeight = FontWeight.Bold,
                    letterSpacing = 0.8.sp,
                    color = if (chokeOn) t.onAccent else t.text2,
                )
            }
        }
    }
}

// ── Buttons — primary (solid accent) / ghost (outline) ────────────────────────
@Composable
fun Ep133PrimaryButton(
    label: String,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
    onClick: () -> Unit,
) {
    val t = LocalEP133Tokens.current
    Box(
        modifier = modifier
            .clip(Radius)
            .background(if (enabled) t.accent else t.rule, Radius)
            .clickable(enabled = enabled) { onClick() }
            .padding(vertical = 11.dp, horizontal = 16.dp),
        contentAlignment = Alignment.Center,
    ) {
        Text(
            label.uppercase(),
            color = if (enabled) t.onAccent else t.text3,
            fontWeight = FontWeight.Bold,
            fontSize = 13.sp,
            letterSpacing = 0.4.sp,
        )
    }
}

@Composable
fun Ep133GhostButton(
    label: String,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
    onClick: () -> Unit,
) {
    val t = LocalEP133Tokens.current
    Box(
        modifier = modifier
            .clip(Radius)
            .border(1.dp, t.rule, Radius)
            .clickable(enabled = enabled) { onClick() }
            .padding(vertical = 11.dp, horizontal = 16.dp),
        contentAlignment = Alignment.Center,
    ) {
        Text(
            label.uppercase(),
            color = if (enabled) t.text else t.text3.copy(alpha = 0.5f),
            fontWeight = FontWeight.Bold,
            fontSize = 13.sp,
            letterSpacing = 0.4.sp,
        )
    }
}

// ── Mono stat readout — LABEL over a big mono VALUE, on an inset panel ─────────
@Composable
fun Ep133StatReadout(label: String, value: String, modifier: Modifier = Modifier) {
    val t = LocalEP133Tokens.current
    Column(
        modifier = modifier
            .clip(Radius)
            .background(t.inset, Radius)
            .border(1.dp, t.rule, Radius)
            .padding(vertical = 9.dp, horizontal = 11.dp),
        verticalArrangement = Arrangement.spacedBy(3.dp),
    ) {
        Text(label.uppercase(), color = t.text3, fontFamily = Mono, fontSize = 8.5.sp, letterSpacing = 1.2.sp)
        Text(value, color = t.text, fontFamily = Mono, fontSize = 15.sp, fontWeight = FontWeight.Bold)
    }
}

// ── Confirm dialog — themed AlertDialog with an accent confirm + plain cancel ─
/**
 * A faceplate-themed confirmation dialog: bold [title], body [message], an accent [confirmLabel]
 * action, and a [dismissLabel] cancel. Used for destructive/irreversible actions (clear, restore).
 */
@Composable
fun Ep133ConfirmDialog(
    title: String,
    message: String,
    confirmLabel: String,
    onConfirm: () -> Unit,
    onDismiss: () -> Unit,
    modifier: Modifier = Modifier,
    dismissLabel: String = "Cancel",
) {
    val t = LocalEP133Tokens.current
    AlertDialog(
        onDismissRequest = onDismiss,
        modifier = modifier,
        containerColor = t.panel,
        titleContentColor = t.text,
        textContentColor = t.text2,
        shape = PanelRadius,
        title = { Text(title, fontWeight = FontWeight.Bold) },
        text = { Text(message) },
        confirmButton = {
            TextButton(
                onClick = onConfirm,
                colors = ButtonDefaults.textButtonColors(contentColor = t.accent),
            ) { Text(confirmLabel, fontWeight = FontWeight.Bold) }
        },
        dismissButton = {
            TextButton(
                onClick = onDismiss,
                colors = ButtonDefaults.textButtonColors(contentColor = t.text2),
            ) { Text(dismissLabel) }
        },
    )
}

File: iOSApp/EP133SampleTool/UI/Theme/EP133Components.swift
import SwiftUI

/// SwiftUI port of `AndroidApp/.../ui/theme/EP133Components.kt`.
///
/// The EP-133 redesign component kit — the SwiftUI mirror of the Claude Design system's component
/// sheet (pads, chips, group selector, status dots, buttons, section labels, app chrome). Every
/// redesigned screen composes from these so the look stays consistent. All colors come from
/// `\.ep133Tokens`; the hardware aesthetic (flat faceplate, hard ~2pt corners, mono labels) is
/// intentionally NOT native chrome — system components are used only where they fit (forms,
/// sliders).

/// Small hard-corner radius used across the faceplate UI.
private let radius: CGFloat = 3

/// Mono labels/codes (the design uses JetBrains Mono; the system monospaced design — SF Mono —
/// is the safe on-device fallback).
private func mono(_ size: CGFloat, _ weight: Font.Weight = .regular) -> Font {
    .system(size: size, weight: weight, design: .monospaced)
}

private extension View {
    /// Compose's `.clip(Radius).background(color, Radius).border(1.dp, border, Radius)` stack.
    func ep133Panel(fill: Color, border: Color? = nil) -> some View {
        background(RoundedRectangle(cornerRadius: radius).fill(fill))
            .overlay {
                if let border {
                    RoundedRectangle(cornerRadius: radius).strokeBorder(border, lineWidth: 1)
                }
            }
            .clipShape(RoundedRectangle(cornerRadius: radius))
            .contentShape(RoundedRectangle(cornerRadius: radius))
    }
}

// ── Section label — uppercase mono eyebrow over a block ───────────────────────
struct EP133SectionLabel: View {
    @Environment(\.ep133Tokens) private var t
    let text: String

    var body: some View {
        Text(text.uppercased())
            .font(mono(9, .medium))
            .tracking(1.6)
            .foregroundStyle(t.text3)
    }
}

// ── Status dot (teal=ok, accent=active, text3=idle) ───────────────────────────
struct EP133StatusDot: View {
    let color: Color
    var size: CGFloat = 8

    var body: some View {
        Circle().fill(color).frame(width: size, height: size)
    }
}

// ── Pad cell ──────────────────────────────────────────────────────────────────
enum PadState { case empty, loaded, pressed, inScale }

/// A rubber pad cell: faceplate-black face, hard corners, mono id + name. `state` drives the id
/// color and the pressed glow / in-scale teal hairline, matching the design's pad variants.
struct EP133Pad: View {
    @Environment(\.ep133Tokens) private var t
    let id: String
    let name: String
    let state: PadState
    var onClick: (() -> Void)? = nil

    var body: some View {
        let idColor: Color = switch state {
        case .empty: Color(argb: 0xFF5D5E5F)
        case .loaded: Color(argb: 0xFFC9CACB)
        case .pressed: t.accent
        case .inScale: t.live
        }
        let border: Color = switch state {
        case .inScale: t.live
        case .pressed: t.accent
        default: t.padEdge
        }
        let face = state == .pressed ? t.padTop : t.padFace

        VStack(alignment: .leading, spacing: 3) {
            Text(id)
                .font(mono(11, .bold))
                .tracking(0.3)
                .foregroundStyle(idColor)
            Text(name.uppercased())
                .font(mono(8))
                .tracking(0.3)
                .foregroundStyle(t.padName)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(.vertical, 11)
        .padding(.horizontal, 8)
        .ep133Panel(fill: face, border: border)
        .onTapGesture { onClick?() }
    }
}

// ── Group selector chip (A / B / C / D, etc.) ─────────────────────────────────
struct EP133GroupChip: View {
    @Environment(\.ep133Tokens) private var t
    let label: String
    let selected: Bool
    var subLabel: String? = nil
    let onClick: () -> Void

    var body: some View {
        Button(action: onClick) {
            VStack(spacing: 1) {
                Text(label)
                    .font(mono(12, .bold))
                    .foregroundStyle(selected ? t.onAccent : t.text2)
                if let subLabel {
                    Text(subLabel)
                        .font(mono(7.5, .bold))
                        .tracking(0.8)
                        .foregroundStyle(selected ? t.onAccent.opacity(0.75) : t.text3)
                }
            }
            .frame(maxWidth: .infinity)
            .padding(.vertical, subLabel != nil ? 6 : 9)
            .padding(.horizontal, 14)
            .ep133Panel(fill: selected ? t.accent : t.inset, border: selected ? t.accent : t.rule)
        }
        .buttonStyle(.plain)
    }
}

// ── Group + choke bar — the shared A/B/C/D picker + choke toggle ───────────────
/// The A/B/C/D group picker plus the CHOKE GROUP toggle, as one block. Both the Loop Chopper and
/// the Kit Builder target a group and can choke it, so they share this control (identical placement
/// and styling) rather than each rolling its own. The choke toggle writes `sound.mutegroup` so pads
/// in the group cut each other off.
///
/// Generic over `Channel` (Android binds directly to `PadChannel`) so the component kit doesn't
/// depend on the domain layer; pass the channel list and a `label` closure.
struct EP133GroupChokeBar<Channel: Hashable>: View {
    @Environment(\.ep133Tokens) private var t
    let channels: [Channel]
    let group: Channel
    let onGroupChange: (Channel) -> Void
    let chokeOn: Bool
    let onChokeChange: (Bool) -> Void
    var label: (Channel) -> String
    var tagFor: ((Channel) -> String?)? = nil
    var testTagFor: ((Channel) -> String)? = nil

    var body: some View {
        VStack(spacing: 7) {
            // A | B | C | D — one chip per group, equal width, each tagged with its designation.
            HStack(spacing: 6) {
                ForEach(channels, id: \.self) { ch in
                    EP133GroupChip(
                        label: label(ch),
                        selected: group == ch,
                        subLabel: tagFor?(ch) ?? nil,
                        onClick: { onGroupChange(ch) }
                    )
                    .accessibilityIdentifier(testTagFor?(ch) ?? "")
                }
            }
            // Choke toggle — full-width row, tap anywhere to flip.
            HStack {
                VStack(alignment: .leading, spacing: 0) {
                    Text("CHOKE GROUP")
                        .font(mono(10.5, .bold))
                        .tracking(0.6)
                        .foregroundStyle(t.text)
                    Text("pads in the group cut each other off")
                        .font(mono(8.5))
                        .foregroundStyle(t.text3)
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                Text(chokeOn ? "ON" : "OFF")
                    .font(mono(10, .bold))
                    .tracking(0.8)
                    .foregroundStyle(chokeOn ? t.onAccent : t.text2)
                    .padding(.horizontal, 13)
                    .padding(.vertical, 5)
                    .ep133Panel(fill: chokeOn ? t.accent : t.chrome)
            }
            .padding(.horizontal, 12)
            .padding(.vertical, 9)
            .ep133Panel(fill: t.inset, border: t.rule)
            .onTapGesture { onChokeChange(!chokeOn) }
            .accessibilityIdentifier(TestTags.GROUP_CHOKE_TOGGLE)
        }
    }
}

// ── Chips — filter (base) and action (solid) + live badge ─────────────────────
struct EP133Chip: View {
    @Environment(\.ep133Tokens) private var t
    let label: String
    var selected: Bool = false
    var onClick: (() -> Void)? = nil

    var body: some View {
        Text(label.uppercased())
            .font(mono(9.5, .medium))
            .tracking(0.6)
            .foregroundStyle(selected ? t.onAccent : t.text2)
            .padding(.vertical, 6)
            .padding(.horizontal, 10)
            .ep133Panel(fill: selected ? t.accent : .clear, border: selected ? t.accent : t.rule)
            .onTapGesture { onClick?() }
    }
}

struct EP133LiveBadge: View {
    @Environment(\.ep133Tokens) private var t
    var label: String = "LIVE"

    var body: some View {
        HStack(spacing: 5) {
            EP133StatusDot(color: t.live, size: 6)
            Text(label.uppercased())
                .font(mono(9, .bold))
                .tracking(0.8)
                .foregroundStyle(t.liveInk)
        }
        .padding(.vertical, 5)
        .padding(.horizontal, 8)
        .ep133Panel(fill: t.live.opacity(0.14))
    }
}

// ── Buttons — primary (solid accent) / ghost (outline) ────────────────────────
struct EP133PrimaryButton: View {
    @Environment(\.ep133Tokens) private var t
    let label: String
    var enabled: Bool = true
    let onClick: () -> Void

    var body: some View {
        Button(action: onClick) {
            Text(label.uppercased())
                .font(.system(size: 13, weight: .bold))
                .tracking(0.4)
                .foregroundStyle(enabled ? t.onAccent : t.text3)
                .frame(maxWidth: .infinity)
                .padding(.vertical, 11)
                .padding(.horizontal, 16)
                .ep133Panel(fill: enabled ? t.accent : t.rule)
        }
        .buttonStyle(.plain)
        .disabled(!enabled)
    }
}

struct EP133GhostButton: View {
    @Environment(\.ep133Tokens) private var t
    let label: String
    var enabled: Bool = true
    let onClick: () -> Void

    var body: some View {
        Button(action: onClick) {
            Text(label.uppercased())
                .font(.system(size: 13, weight: .bold))
                .tracking(0.4)
                .foregroundStyle(enabled ? t.text : t.text3.opacity(0.5))
                .frame(maxWidth: .infinity)
                .padding(.vertical, 11)
                .padding(.horizontal, 16)
                .ep133Panel(fill: .clear, border: t.rule)
        }
        .buttonStyle(.plain)
        .disabled(!enabled)
    }
}

// ── Mono stat readout — LABEL over a big mono VALUE, on an inset panel ─────────
struct EP133StatReadout: View {
    @Environment(\.ep133Tokens) private var t
    let label: String
    let value: String

    var body: some View {
        VStack(alignment: .leading, spacing: 3) {
            Text(label.uppercased())
                .font(mono(8.5))
                .tracking(1.2)
                .foregroundStyle(t.text3)
            Text(value)
                .font(mono(15, .bold))
                .foregroundStyle(t.text)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(.vertical, 9)
        .padding(.horizontal, 11)
        .ep133Panel(fill: t.inset, border: t.rule)
    }
}

// ── App chrome — header + body + bottom tab nav ───────────────────────────────
/// The bottom-nav destinations, in order (mirrors `NavRoute` in EP133App.kt).
enum EP133Tab: CaseIterable {
    case pads, kit, projects, device

    var label: String {
        switch self {
        case .pads: "PADS"
        case .kit: "SAMPLES"
        case .projects: "PROJ"
        case .device: "DEVICE"
        }
    }

    var testTag: String {
        switch self {
        case .pads: TestTags.NAV_PADS
        case .kit: TestTags.NAV_KIT
        case .projects: TestTags.NAV_PROJECTS
        case .device: TestTags.NAV_DEVICE
        }
    }
}

/// App shell chrome, mirroring `EP133App` in EP133App.kt: faceplate header (tappable SKU badge +
/// screen title + theme-mode toggle + connection dot) wrapping the screen `content`, with the mono
/// bottom tab nav. Theme/SKU state lives in the caller (AppShell) so the header controls can
/// re-theme the whole app.
struct EP133Scaffold<Content: View>: View {
    @Environment(\.ep133Tokens) private var t
    let title: String
    let connected: Bool
    let currentTab: EP133Tab
    let onTab: (EP133Tab) -> Void
    var sku: EP133Sku = .ep133
    var onSkuToggle: (() -> Void)? = nil
    var themeGlyph: String? = nil
    var onThemeToggle: (() -> Void)? = nil
    @ViewBuilder let content: () -> Content

    var body: some View {
        VStack(spacing: 0) {
            // ── Faceplate header ──
            HStack(spacing: 0) {
                // Tap the badge to switch SKU theme (EP-133 orange ↔ EP-1320 rust).
                Button(action: { onSkuToggle?() }) {
                    Text(sku == .ep133 ? "EP·133" : "EP·1320")
                        .font(mono(11, .bold))
                        .foregroundStyle(t.onAccent)
                        .padding(.horizontal, 7)
                        .padding(.vertical, 4)
                        .ep133Panel(fill: t.accent)
                }
                .buttonStyle(.plain)
                .accessibilityIdentifier(TestTags.HEADER_SKU_BADGE)
                .accessibilityLabel(
                    sku == .ep133
                        ? "Switch to the EP-1320 color theme"
                        : "Switch to the EP-133 color theme"
                )
                // Announce the currently selected SKU (VoiceOver reads it as the control's value,
                // and it's the stable read the UI tests assert on since the Button collapses its
                // child Text out of the accessibility tree).
                .accessibilityValue(sku == .ep133 ? "EP·133" : "EP·1320")
                Text(title.uppercased())
                    .font(mono(11, .semibold))
                    .tracking(1.1)
                    .foregroundStyle(t.text2)
                    .padding(.leading, 9)
                    .frame(maxWidth: .infinity, alignment: .leading)
                    .accessibilityIdentifier(TestTags.HEADER_TITLE)
                // Tap to cycle theme: follow system → light → dark.
                if let themeGlyph {
                    Button(action: { onThemeToggle?() }) {
                        Text(themeGlyph)
                            .font(.system(size: 14))
                            .foregroundStyle(t.text2)
                            .padding(.horizontal, 8)
                            .padding(.vertical, 3)
                    }
                    .buttonStyle(.plain)
                    .accessibilityIdentifier(TestTags.HEADER_THEME_TOGGLE)
                    Spacer().frame(width: 8)
                }
                let dot = connected ? t.live : t.text3
                HStack(spacing: 7) {
                    EP133StatusDot(color: dot, size: 8)
                    Text(connected ? "CONNECTED" : "NO DEVICE")
                        .font(mono(10))
                        .tracking(0.7)
                        .foregroundStyle(dot)
                }
                .accessibilityElement(children: .combine)
                .accessibilityIdentifier(TestTags.HEADER_CONNECTION)
            }
            .padding(.horizontal, 15)
            .padding(.vertical, 13)
            .background(t.panel2)
            // ── Screen body ──
            content()
                .frame(maxWidth: .infinity, maxHeight: .infinity)
            // ── Faceplate bottom tab nav ──
            HStack(spacing: 0) {
                ForEach(EP133Tab.allCases, id: \.self) { tab in
                    let active = tab == currentTab
                    Text(tab.label)
                        .font(mono(8.5, active ? .bold : .medium))
                        .tracking(0.3)
                        .foregroundStyle(active ? t.accent : t.text3)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 11)
                        .contentShape(Rectangle())
                        .onTapGesture { onTab(tab) }
                        .accessibilityIdentifier(tab.testTag)
                        .accessibilityAddTraits(active ? [.isButton, .isSelected] : .isButton)
                }
            }
            // Nav background bleeds under the home indicator (Android's navigationBarsPadding
            // inside the panel2 background); the status-bar area above shows bg, matching
            // Android's statusBarsPadding on the bg column.
            .background(t.panel2.ignoresSafeArea(edges: .bottom))
        }
        .background(t.bg.ignoresSafeArea())
    }
}
